lib/ds/hashmap.flow

```diff
@@ -102,26 +102,44 @@ makeHashMapOfVolume(hash : (?) -> int, volume : int, loadFactor : double) -> Has
 }
 
 lookupHashMap(hashMap : HashMap<?, ??>, key : ?) -> Maybe<??> {
-	lookupTree(^(hashMap.data[hashMapIndex(hashMap, key)]), key)
+	index = hashMapIndex(hashMap, key);
+	if (existsIndex(hashMap.data, index)) {
+		lookupTree(^(hashMap.data[index]), key)
+	} else {
+		None()
+	}
 }
 
 lookupHashMapDef(hashMap : HashMap<?, ??>, key : ?, defaultValue : ??) -> ?? {
-	lookupTreeDef(^(hashMap.data[hashMapIndex(hashMap, key)]), key, defaultValue)
+	index = hashMapIndex(hashMap, key);
+	if (existsIndex(hashMap.data, index)) {
+		lookupTreeDef(^(hashMap.data[index]), key, defaultValue)
+	} else {
+		defaultValue
+	}
 }
 
 containsKeyHashMap(hashMap : HashMap<?, ??>, key : ?) -> bool {
+	index = hashMapIndex(hashMap, key);
+	if (existsIndex(hashMap.data, index)) {
+		containsKeyTree(^(hashMap.data[index]), key)
+	} else {
+		false
+	}
 	containsKeyTree(^(hashMap.data[hashMapIndex(hashMap, key)]), key)
```

Comment:
Should be removed? Consider && instead of if

Reply:
sorry. fixed